### PR TITLE
HTTP stub helpers

### PR DESCRIPTION
**Pull request by pengwynn**
Inspired by [Octokit.rb's stub helpers][helpers], this is a work-in-progress proof-of-concept for removing some noise from our tests. We can have our :cake: and eat it, too, because you can always do the more verbose approach if you need to.

Before we get too far, I wanted to see what @jingweno thinks of the approach.

/cc @obsc @dhruvsinghal @dhs252 

[helpers]: https://github.com/octokit/octokit.rb/blob/568745f0a612971207dd4d05ccbf0af821bdbb36/spec/helper.rb#L121-L170

### CODE REVIEW

**Comment by jingweno** [comment]
@pengwynn :+1: Good refactoring! So much DRYer.

File: octokit/octokit_test.go [hunk 1]
```diff
@@ -122,3 +122,24 @@ func loadFixture(f string) string {
 	c, _ := ioutil.ReadFile(p)
 	return string(c)
 }
+
+func stubGet(t *testing.T, path, fixture string, params map[string]string) {
+	stubRequest(t, "GET", path, fixture, params)
+}
+
+func stubRequest(t *testing.T, method string, path string, fixture string, params map[string]string) {
```
Comment (jingweno) [inline octokit/octokit_test.go]: Could prob. pass a func at the last for assertion or optional manipulation as well:

```go
func stubRequest(t *testing.T, method string, path string, fixture string, params map[string]string, f func(w http.ResponseWriter, r *http.Request) {
  ...

  mux.HandleFunc(path, func(w http.ResponseWriter, r *http.Request) {
       if f != nil {
          f(w, r)
       } 
  }
}
```

Comment (pengwynn) [inline octokit/octokit_test.go]: Oh good call. I like that idea.

Comment (obsc) [inline octokit/octokit_test.go]: I really like that idea as well.

**Pull request merged**